src/Symfony/Component/DependencyInjection/Compiler/ResolveStaticConstructorPass.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+class ResolveStaticConstructorPass extends AbstractRecursivePass
```

Comment:
We don't need a new pass to handle this - and we don't need a new property on Definition: `setFactory([null, 'theMethod'])` is enough to hold the needed state.

Reply:
I removed the pass and updated Defintion to remove the new property 👍